api/models/reviewDao.js

```diff
@@ -0,0 +1,157 @@
+const dbDataSource = require("./dataSource");
+
+const createReview = async (userId, rating, comment, movieId) => {
+  await dbDataSource.query(`set FOREIGN_KEY_CHECKS = 0`);
+  await dbDataSource.query(
+    `
+    INSERT INTO movie_reviews (
+      users_id, rating, comment, movies_id
+    ) VALUES (
+      ?, ?, ?, ?
+    )
+    `,
+    [userId, rating, comment, movieId]
+  );
+  await dbDataSource.query(`set FOREIGN_KEY_CHECKS = 1`);
+
+  let totalReview = await dbDataSource.query(
+    `
+    SELECT COUNT(id) as id
+    FROM movie_reviews
+    WHERE movies_id = ?
+    `,
+    [movieId]
+  );
+  totalReview = totalReview[0].id;
+
+  let averageRating = await dbDataSource.query(
+    `
+    SELECT SUM(rating) as average_rating
+    FROM movie_reviews
+    WHERE movies_id = ?
+    `,
+    [movieId]
+  );
+  averageRating = averageRating[0].average_rating;
+
+  await dbDataSource.query(
+    `
+    UPDATE movies
+    SET average_rating = ?
+    WHERE id = ?
+    `,
+    [averageRating / totalReview, movieId]
+  );
+};
+
+const getReview = async (movieId) => {
+  const result = await dbDataSource.query(
+    `
+    SELECT rating, comment, u.nickname
+    FROM movie_reviews as mr
+    JOIN users as u
+    ON mr.users_id = u.id
+    WHERE mr.movies_id = ?
+    `,
+    [movieId]
+  );
+  return result;
+};
+
+const updateReview = async (comment, checkReview) => {
+  const result = await dbDataSource.query(
+    `
+    UPDATE movie_reviews 
+    SET comment = ?
+    WHERE movie_reviews.id = ?
+    `,
+    [comment, checkReview]
+  );
+  return result;
+};
+
+const deleteReview = async (movieId, checkReview) => {
+  await dbDataSource.query(
+    `
+    DELETE FROM movie_reviews
+    WHERE movie_reviews.id = ?
+    `,
+    [checkReview]
+  );
+  let totalReview = await dbDataSource.query(
+    `
+    SELECT COUNT(id) as id
+    FROM movie_reviews
+    WHERE movies_id = ?
+    `,
+    [movieId]
+  );
+  totalReview = totalReview[0].id;
+
+  if (totalReview == 0) {
+    await dbDataSource.query(
+      `
+      UPDATE movies
+      SET average_rating = 0
+      WHERE id = ?
+      `,
+      [movieId]
+    );
+  } else {
+    let averageRating = await dbDataSource.query(
+      `
+      SELECT SUM(rating) as average_rating
+      FROM movie_reviews
+      WHERE movies_id = ?
+      `,
+      [movieId]
+    );
+    averageRating = averageRating[0].average_rating;
+
+    await dbDataSource.query(
+      `
+      UPDATE movies
+      SET average_rating = ?
+      WHERE id = ?
+      `,
+      [averageRating / totalReview, movieId]
+    );
+  }
```

Comment:
```suggestion
  const [result] = await dbDataSource.query(
    `
    SELECT COUNT(id) as totalReviews, SUM(rating) as totalRating
    FROM movie_reviews
    WHERE movies_id = ?
    `,
    [movieId]
  );

  const totalReview = result.totalReviews;
  const averageRating = totalReview ? result.totalRating / totalReview : 0;

  await dbDataSource.query(
    `
    UPDATE movies
    SET average_rating = ?
    WHERE id = ?
    `,
    [averageRating, movieId]
  );
```
위에서 작성해주신 코드를 복잡한 로직없이 삼항연산자를 활용하여 간소화한 버전입니다. 천천히 읽어보시고 적용시켜주세요.